python/pyspark/sql/streaming/stateful_processor_util.py

```diff
@@ -25,3 +25,4 @@ class TransformWithStateInPandasFuncMode(Enum):
     PROCESS_DATA = 1
     PROCESS_TIMER = 2
     COMPLETE = 3
+    PRE_INIT = 4
```

Comment:
This is a stateful processor handle state introduced only for internal use. It will be passed into our UDF `transformWithStateUDF` in `group_ops.py` as an internal parameter and we will use this parameter to decide whether we are going to process data, timer, or if we are at pre-init stage. The public exposed StatefulProcessorHandleState will only be set on the JVM.

Reply:
Ah ok - could we add this as a comment in the code too ?

Reply:
>could we add this as a comment in the code too ?

Yeah this is documented as class comment here: https://github.com/apache/spark/blob/master/python/pyspark/sql/streaming/stateful_processor_util.py#L20